infra/src/main/kotlin/fi/fta/geoviite/infra/localization/LocalizationService.kt

```diff
@@ -1,18 +1,42 @@
 package fi.fta.geoviite.infra.localization
 
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonValue
 import com.fasterxml.jackson.databind.json.JsonMapper
-import fi.fta.geoviite.infra.error.LocalizationParams
 import fi.fta.geoviite.infra.util.LocalizationKey
 import org.springframework.beans.factory.annotation.Value
 import org.springframework.stereotype.Service
 import java.io.File
 import java.util.concurrent.ConcurrentHashMap
 
+private val LOCALIZATION_PARAMS_KEY_REGEX = Regex("[a-zA-Z0-9_\\s\\-]*")
+private val LOCALIZATION_PARAMS_PLACEHOLDER_REGEX = Regex("\\{\\{[a-zA-Z0-9_\\s\\-]*\\}\\}")
+
+data class LocalizationParams @JsonCreator(mode = JsonCreator.Mode.DELEGATING)
+constructor(@JsonValue val params: Map<String, Any?>) {
+    constructor(vararg params: Pair<String, Any?>) : this(mapOf(*params))
+
+    init {
+        require(params.none { LOCALIZATION_PARAMS_KEY_REGEX.matchEntire(it.key) == null }) {
+            "There are localization keys that do not with the localization regex. Keys are ${
```

Comment:
"... that do not with the ..." Kuulostaisi siltä että tuosta puuttuu joku verbi välistä?

Reply:
😬